### PR TITLE
Add structured suggestions for various "use" suggestions

### PR DESCRIPTION
**Pull request by oli-obk**
r? @petrochenkov

### CODE REVIEW

File: src/test/ui/issue-35976.stderr [hunk 1]
```diff
@@ -4,8 +4,10 @@ error: the `wait` method cannot be invoked on a trait object
 24 |     arg.wait();
    |         ^^^^
    |
```
Comment (kennytm) [inline src/test/ui/issue-35976.stderr]: This line needs to be deleted due to recent contributions:

```diff
 error: the `wait` method cannot be invoked on a trait object
   --> $DIR/issue-35976.rs:24:9
    |
 24 |     arg.wait();
    |         ^^^^
-   |
 help: another candidate was found in the following trait, perhaps add a `use` for it:
    |
 11 | use private::Future;
    |
 
 error: aborting due to previous error
```

Comment (oli-obk) [reply src/test/ui/issue-35976.stderr]: done

**Comment by oli-obk** [comment]
spurious?

```
---- changing_bin_features_caches_targets stdout ----
	running `C:\projects\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\release\cargo.exe run`
running `C:\projects\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\release\cargo.exe run --features foo`
thread 'changing_bin_features_caches_targets' panicked at '
Expected: execs
    but: exited with exit code: 101
--- stdout
--- stderr
   Compiling foo v0.0.1 (file:///C:/projects/rust/build/x86_64-pc-windows-msvc/stage2-tools/x86_64-pc-windows-msvc/cit/t3/foo)
error: failed to link or copy `C:\projects\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\cit\t3\foo\target\debug\deps\foo-f76c704c72dc1115.exe` to `C:\projects\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\cit\t3\foo\target\debug\foo.exe`
Caused by:
  Access is denied. (os error 5)
', C:\Users\appveyor\.cargo\registry\src\github.com-1ecc6299db9ec823\hamcrest-0.1.1\src\core.rs:31:12
note: Run with `RUST_BACKTRACE=1` for a backtrace.
---- changing_bin_paths_common_target_features_caches_targets stdout ----
	running `C:\projects\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\release\cargo.exe run`
running `C:\projects\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\release\cargo.exe clean -p a`
running `C:\projects\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\release\cargo.exe run`
thread 'changing_bin_paths_common_target_features_caches_targets' panicked at '
Expected: execs
    but: exited with exit code: 101
--- stdout
--- stderr
   Compiling a v0.0.1 (file:///C:/projects/rust/build/x86_64-pc-windows-msvc/stage2-tools/x86_64-pc-windows-msvc/cit/t2/foo/a)
error: failed to link or copy `C:\projects\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\cit\t2\foo\./target\debug\deps\a-a478fe7568e0eb90.exe` to `C:\projects\rust\build\x86_64-pc-windows-msvc\stage2-tools\x86_64-pc-windows-msvc\cit\t2\foo\./target\debug\a.exe`
Caused by:
  Access is denied. (os error 5)
', C:\Users\appveyor\.cargo\registry\src\github.com-1ecc6299db9ec823\hamcrest-0.1.1\src\core.rs:31:12
failures:
    changing_bin_features_caches_targets
    changing_bin_paths_common_target_features_caches_targets
```

**Comment by kennytm** [comment]
@bors retry

Not sure, but feels spurious

**Pull request merged**